server/controllers/volunteerController.js

```diff
@@ -359,10 +337,49 @@ let VolunteerController = function (volunteerRepository, shiftRepository, userRe
       .then(shifts => res.status(200).json({message: "Success!", shifts, count: shifts.length}))
       .catch(err => res.status(500).json({message: err}));
   };
+
+  this.invite = function (req, res) {
+    if (!req.user.isAdmin) {
+      res.status(401).json({message: "Only admins can invite volunteers"});
+      return;
+    }
+    if (!req.user.email) {
+      res.status(400).json({message: "No email has been specified"});
+    }
+    volunteerRepository.getByEmail(req.body.email)
+      .then(volunteer => {
+        if (volunteer) {
+          res.status(400).json({message: "Volunteer already has an account"});
+          return;
+        }
+        return invitationTokenRepository.getByEmail(req.body.email);
+      })
+      .then(invitation => {
+        if (invitation && moment().isBefore(invitation.expires)) {
+          res.status(400).json({message: "Volunteer with that email has already been invited!"});
+          return;
+        }
+        const token = crypto.randomBytes(16).toString('hex');
+        const expires = moment().add(1, 'days').format();
+        return invitationTokenRepository.add(req.body.email, token, expires)
+          .then(() => {
+            const emailClient = new EmailClient(emailClientTypes.NOREPLY);
+            emailClient.send(req.body.email,
+              "Invitation to City Harvest",
+              `Hello from Mobilise,\n
+          You have been invited to join City Harvest by ${req.user.firstName}.\n
+          Please click the following link to sign-up to Mobilise, the home of volunteering at City Harvest.\n\n
+          ${process.env.WEB_URL}/sign-up?token=${token}\n\n
+          This link will expire in 24 hours.`);
+            res.status(200).json({message: "Success! Invitation has been sent."});
+          })
```

Comment:
This promise does not get caught - is this intentional?

Reply:
Ah wasn't aware it was a promise, will fix now

Reply:
Done